File: examples/py/websocket-playground.py
import os
import sys
import pprint
import traceback

pp = pprint.PrettyPrinter(depth=6)

root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(root + '/python')
# import ccxt  # noqa: E402
import ccxt.async as ccxt  # noqa: E402
import asyncio  # noqa: E402

loop = asyncio.get_event_loop()
# import txaio
# txaio.start_logging(level='debug')


async def main():
    if len(sys.argv) <= 3:
        print('python ' + __file__ + ' exchange event symbol [params]')
        sys.exit(-1)

    exchange_id = sys.argv[1]
    apiKey = 'API_KEY' in os.environ and os.environ['API_KEY'] or ''
    secret = 'SECRET' in os.environ and os.environ['SECRET'] or ''
    event = sys.argv[2]
    symbol = sys.argv[3]
    params = {}
    for i in range(4, len(sys.argv)):
        parts = sys.argv[i].split(':',2)
        params[parts[0]] = parts[1]


    exchange = getattr(ccxt, exchange_id)({
        "apiKey": apiKey,
        "secret": secret,
        "enableRateLimit": True,
        'verbose': True,
        'timeout': 5 * 1000,
        # 'wsproxy': 'http://185.93.3.123:8080/',
    })

    @exchange.on('err')
    def websocket_error(err, conxid):  # pylint: disable=W0612
        print(type(err).__name__ + ":" + str(err))
        traceback.print_tb(err.__traceback__)
        traceback.print_stack()
        loop.stop()

    @exchange.on(event)
    def websocket_ob(symbol, data):  # pylint: disable=W0612
        print(event + " received from: " + symbol)
        pp.pprint(data)
        sys.stdout.flush()

    sys.stdout.flush()

    await exchange.websocket_subscribe(event, symbol, params)
    print("subscribed: " + symbol)

#loop.run_until_complete(main())
try:
    asyncio.ensure_future(main())
    loop.run_forever()
except KeyboardInterrupt:
    pass
finally:
    print("Closing Loop")
    loop.close()
# loop.stop()
# loop.close()
print("after complete")

File: examples/php/websocket-playground.php
<?php 
$root = dirname (dirname (dirname (__FILE__)));

include $root . '/ccxt.php';
include $root . '/php/Exchange.php';
include $root . '/php/cex.php';

require __DIR__ . '/../../vendor/autoload.php';

if ($argc > 3) {
    $id = $argv[1];
    $apiKey =  isset($_ENV["API_KEY"])?$_ENV["API_KEY"]:'';
    $secret =  isset($_ENV["SECRET"])?$_ENV["SECRET"]:'';
    $event = $argv[2];
    $symbol = $argv[3];
    $params = array();
    for ($i=4;$i<$argc;$i++){
        $parts = explode(':',$argv[$i],2);
        $params[$parts[0]] = $parts[1];
    }
    if ($argc > 4){
        preg_match_all('~^var\s+([^=]+?)\s*=\s*(.+?)\s*;?\s*$~imu', $argv[4], $matchesAll, PREG_SET_ORDER);
        foreach ($matchesAll as $matches) {
            $params[$matches[1]] = $matches[2];
        }
    }
    
} else {
    echo 'php '. __FILE__ . ' exchange-id event symbol [params] ...\n';
    exit(-1);
}

$loop = React\EventLoop\Factory::create();

$cl = 'ccxt\\' . $id;

$exchange = new  $cl (array(
    'apiKey' => $apiKey,
    'secret' => $secret,
    'enableRateLimit' => true,
    'verbose' => true,
    'timeout' => 5 * 1000,
    'react_loop' => $loop
));

$exchange->on ('err', function ($err, $conxid) use ($exchange){
    echo ($err);
    $exchange->websocketClose();
});

$exchange->on ($event, function ($symbol, $data) use($event) {
    echo ($event. " received from: " . $symbol);
    print_r ($data);
    echo ("\n");
});


$exchange->websocket_subscribe ($event, $symbol, $params);
echo('subscribed ' . $symbol);
$loop->run();